crates/starfish/core/src/test_dag_builder.rs

```diff
@@ -575,19 +597,23 @@ impl<'a> LayerBuilder<'a> {
             } else if self.min_ancestor_links {
                 self.configure_min_parent_links()
             } else if self.no_leader_link {
-                self.configure_no_leader_links(authorities.clone(), round)
-            } else if self.skip_ancestor_links.is_some() {
-                self.configure_skipped_ancestor_links(
-                    authorities,
-                    self.skip_ancestor_links.clone().unwrap(),
-                )
+                self.configure_no_leader_links(&authorities, round)
+            } else if let Some(ancestors_to_skip) = self.skip_ancestor_links.clone() {
+                self.configure_skipped_ancestor_links(&authorities, ancestors_to_skip)
             } else {
                 vec![]
             };
 
             // Do not acknowledge transactions in round 0 (genesis).
-            let acknowledgments = if round > 1 && self.fully_linked_acknowledgments {
+            // TODO: check if <= round 1 is correct or if it should be < round 1
```

Comment:
It's correct. Acknowledgements only make sense from round 2 if they acknowledge transactions in round 1.

Reply:
ok, will remove the TODO in cleanup commit